tests/nocopy_test.go

```diff
@@ -0,0 +1,41 @@
+package tests
+
+import (
+	"reflect"
+	"testing"
+	"unsafe"
+
+	"github.com/mailru/easyjson"
+)
+
+// verifies if string pointer belongs to the given buffer or outside of it
+func strBelongsTo(s string, buf []byte) bool {
+	sPtr := (*reflect.StringHeader)(unsafe.Pointer(&s)).Data
+	bufPtr := (*reflect.SliceHeader)(unsafe.Pointer(&buf)).Data
+
+	if bufPtr <= sPtr && sPtr < bufPtr+uintptr(len(buf)) {
+		return true
+	}
+	return false
+}
+
+func TestNocopy(t *testing.T) {
```

Comment:
you can add additional test that testing.AllocsPerRun = 1 (for A only)

Reply:
ok. will add.

Reply:
Added, but honestly I dislike this kind of tests. They ALL gonna start failing when compiler realise that lexer doesn't need to be allocated.